src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1070 @@
+                    rustdoc
+                )
+                .unwrap();
+                rustdoc.clear();
+
+                writeln!(
+                    &mut struct_member_fields_initialization,
+                    "{}: bit_fields::BitRange(std::marker::PhantomData),",
+                    field_ident
+                )
+                .unwrap();
+            }
+            // If we have `<ident>,<punctuation>,<literal>,<,>` it is a bit flag.
+            Some(TokenTree::Punct(punct)) if punct.as_char() == ',' => add_bit_flags(),
+            // If we have `<ident>,<punctuation>,<literal>,<End of stream>` it is a bit flag.
+            None => add_bit_flags(),
+            // If we have `<ident>,<punctuation>,<literal>,<_>` it is an invalid input.
+            Some(_) => macro_error!(field_ident, FIELDS_ERR),
+        }
+        // We skip the punctuation for the next iteration.
+        fields_iter.next();
+    }
+    // If position of last mapped area is before end of underlying data
+    if pos < bits_len.size() {
+        #[rustfmt::skip]
+        display_string.push_str("\
+               ───┐\n\
+            \x20… │\n\
+               ───┤\n\
+            \x20… │\n\
+               ───┤\n\
+            \x20… │\n\
+               ───┘\
+        ");
+    }
+
+    // Convert to token streams
+    #[cfg(feature = "flag_set")]
+    let flag_matching_from_hashset: proc_macro2::TokenStream =
+        flag_matching_from_hashset.parse().unwrap();
+    #[cfg(feature = "field_map")]
+    let field_matching_from_hashmap: proc_macro2::TokenStream =
+        field_matching_from_hashmap.parse().unwrap();
+    #[cfg(feature = "field_map")]
+    let field_setting_hashmap: proc_macro2::TokenStream = field_setting_hashmap.parse().unwrap();
+
+    #[cfg(feature = "flag_set")]
+    let flag_setting_hashset: proc_macro2::TokenStream = flag_setting_hashset.parse().unwrap();
+    #[cfg(feature = "set_theory")]
+    let fields_superset_fn: proc_macro2::TokenStream = fields_superset_fn.parse().unwrap();
+    #[cfg(feature = "set_theory")]
+    let fields_subset_fn: proc_macro2::TokenStream = fields_subset_fn.parse().unwrap();
+    #[cfg(feature = "set_theory")]
+    let fields_disjoint_fn: proc_macro2::TokenStream = fields_disjoint_fn.parse().unwrap();
+    #[cfg(feature = "set_theory")]
+    let fields_intersection_fn: proc_macro2::TokenStream = fields_intersection_fn.parse().unwrap();
+    #[cfg(feature = "set_theory")]
+    let fields_union_fn: proc_macro2::TokenStream = fields_union_fn.parse().unwrap();
+    let struct_doc_table_layout: proc_macro2::TokenStream =
+        struct_doc_table_layout.parse().unwrap();
+    let struct_member_fields: proc_macro2::TokenStream = struct_member_fields.parse().unwrap();
+    let struct_member_fields_initialization: proc_macro2::TokenStream =
+        struct_member_fields_initialization.parse().unwrap();
+
+    #[cfg(feature = "serde")]
+    let visitor = quote::format_ident!("{}Visitor", struct_name);
```

Comment:
What is Visitor?

Reply:
This relates to [Implementing Deserialize](https://serde.rs/impl-deserialize.html).